replacer_test.go

```diff
@@ -0,0 +1,275 @@
+	name, _ := os.Hostname()
+	return name
+}
+
+func getEnvForTest(name string) (string, bool) {
+	val := os.Getenv(name)
+	return val, val != ""
+}
+
+// Tests the Set method by setting some variables and check if they are added to static
+func TestReplacerSet(t *testing.T) {
+	rep := fakeReplacer()
+
+	for _, tc := range []struct {
+		variable string
+		value    string
+	}{
+		{
+			variable: "test1",
+			value:    "val1",
+		},
+		{
+			variable: "asdf",
+			value:    "123",
+		},
+		{
+			variable: "äöü",
+			value:    "öö_äü",
+		},
+		{
+			variable: "with space",
+			value:    "space value",
+		},
+		{
+			variable: "1",
+			value:    "test-123",
+		},
+		{
+			variable: "mySuper_IP",
+			value:    "1.2.3.4",
+		},
+		{
+			variable: "testEmpty",
+			value:    "",
+		},
+	} {
+		rep.Set(tc.variable, tc.value)
+
+		// test if key is added
+		if val, ok := rep.static[tc.variable]; ok {
+			if val != tc.value {
+				t.Errorf("Expectd value '%s' for key '%s' got '%s'", tc.value, tc.variable, val)
+			}
+		} else {
+			t.Errorf("Expectd existing key '%s' found nothing", tc.variable)
+		}
+	}
+
+	// test if all keys are still there (by length)
+	length := len(rep.static)
+	if len(rep.static) != 7 {
+		t.Errorf("Expectd length '%v' got '%v'", 7, length)
+	}
+}
+
+func TestReplacerReplaceAll(t *testing.T) {
+	rep := fakeReplacerFilled()
+
+	for _, tc := range []struct {
+		testInput string
+		expected  string
+	}{
+		{
+			// test vars without space
+			testInput: "{test1}{asdf}{1}",
+			expected:  "val1123test-123",
+		},
+		{
+			// test vars with space
+			testInput: "{test1} {asdf} {1} ",
+			expected:  "val1 123 test-123 ",
+		},
+		{
+			// test with empty val
+			testInput: "{test1} {testEmpty} {asdf} {1} ",
+			expected:  "val1 EMPTY 123 test-123 ",
+		},
+		{
+			// test vars with not finished placeholders
+			testInput: "{te{test1}{as{{df{1}",
+			expected:  "{teval1{as{{dftest-123",
+		},
+		{
+			// test with non existing vars
+			testInput: "{test1} {nope} {1} ",
+			expected:  "val1 {nope} test-123 ",
+		},
+	} {
+		actual := rep.ReplaceAll(tc.testInput, "EMPTY")
+
+		// test if all are replaced as expected
+		if actual != tc.expected {
+			t.Errorf("Expectd '%s' got '%s' for '%s'", tc.expected, actual, tc.testInput)
+		}
+	}
+}
+
+func TestReplacerReplaceAllDefaults(t *testing.T) {
+	rep := NewReplacer()
+	testInput := "{system.hostname} {system.slash} {system.os} {system.arch}"
+	expected := getHostnameForTest() + " " + string(filepath.Separator) + " " + runtime.GOOS + " " + runtime.GOARCH
+
+	// test env.
+	testInput += " {env.GOPATH}"
+	if env, ok := getEnvForTest("GOPATH"); ok {
+		expected += " " + env
+	} else {
+		expected += " "
+	}
+
+	actual := rep.ReplaceAll(testInput, "EMPTY")
+
+	// test if all are replaced as expected
+	if actual != expected {
+		t.Errorf("Expectd '%s' got '%s' for '%s'", expected, actual, testInput)
+	}
+}
+
+func TestReplacerDelete(t *testing.T) {
+	rep := fakeReplacerFilledStatic()
```

Comment:
Again, just inline the value here; no need for a separate function. Less indirection.

Reply:
done